taxon_filter.py

```diff
@@ -554,9 +529,89 @@ def wrapper(inBam, db, outBam, threads, JVMmemory=None):
+                    # this is an unindexed fasta file, we will need to index it
+                    # function should conform to the signature: 
+                    # db_build_command(inputFasta, outputDirectory, outputFilePrefix)
+                    # the function will need to be able to handle lz4, etc.
+                    db_build_command(db, tempDbDir, db_prefix)
+                    db_dir = tempDbDir
+                else:
+                    # this is a tarball with prebuilt indexes
+                    db_dir = util.file.extract_tarball(db, tempDbDir)
+            else:
+                # this is a directory
+                db_dir = db
+            # this directory should have a .nin file
+            hits = list(glob.glob(os.path.join(db_dir, '*.{ext}'.format(ext=db_extension_to_expect))))
+            if len(hits) == 0:
+                raise Exception("The blast database does not appear to a *.{ext} file.".format(ext=db_extension_to_expect))
+            elif len(hits) == 1:
+                db_prefix = hits[0][:-(len('.{ext}'.format(ext=db_extension_to_expect)))]  # remove the '.extension'
+            elif len(hits) >1:
+                db_prefix = os.path.commonprefix(hits).rsplit('.', 1)[0] # remove extension and split-db prefix
+        else:
+            # this is simply a prefix to a bunch of files, not an actual file
+            db_prefix = db
+
+        yield (db_prefix,tempDbDir)
+
+# ========================
+# ***  deplete_bwa  ***
+# ========================
+
+def deplete_bwa_bam(inBam, db, outBam, threads=None, JVMmemory=None):
+    'Use blastn to remove reads that match at least one of the databases.'
+
+    threads = util.misc.sanitize_thread_count(threads)
+
+    with extract_build_or_use_database(db, bwa_build_db, 'bwt', tmp_suffix="-bwa_db_unpack", db_prefix="bwa") as (db_prefix,tempDbDir):
+        with util.file.tempfname('.aligned.sam') as aligned_sam:
+            tools.bwa.Bwa().align_mem_bam(inBam, db_prefix, aligned_sam, threads=threads)
+            tools.samtools.SamtoolsTool().view(['-f0x4'], aligned_sam, outBam)
+            # TODO: consider using Bwa().mem() so the input bam is not broken out by read group
+            # TODO: pipe bwa input directly to samtools process (need to use Bwa().mem() directly, )
+            #       with Popen to background bwa process
+
+def parser_deplete_bwa_bam(parser=argparse.ArgumentParser()):
+    parser.add_argument('inBam', help='Input BAM file.')
+    parser.add_argument('refDbs', nargs='+', help='One or more reference databases for bwa. '
+                         'An ephemeral database will be created if a fasta file is provided.')
+    parser.add_argument('outBam', help='Ouput BAM file with matching reads removed.')
+    #parser.add_argument("--chunkSize", type=int, default=1000000, help='FASTA chunk size (default: %(default)s)')
```

Comment:
The commented out `--chunkSize` and the actual argument `--JVMmemory` can probably both be removed here. This is a nice, Java-free command thus far.

Reply:
Thanks!